base/genimport/importer.go

```diff
@@ -269,14 +272,120 @@ func removeAllFilesInDirExcept(o *Output, dir string, except_list []string) {
 }
 
 func createPluginGoModFile(o *Output, pkgpath string, dir string) string {
+	newLines := [][]byte{
+		[]byte("module gomacro.imports/" + pkgpath + "\n"),
+	}
+
+	if localpath, ok := local(o, pkgpath); ok {
+		newLines = append(newLines, goModReplacementDirectives(o, localpath)...)
+	}
+
+	newGoModText := bytes.Join(newLines, nil)
+
 	gomod := paths.Subdir(dir, "go.mod")
-	err := ioutil.WriteFile(gomod, []byte("module gomacro.imports/"+pkgpath+"\n"), os.FileMode(0644))
+	err := ioutil.WriteFile(gomod, newGoModText, os.FileMode(0o644))
 	if err != nil {
 		o.Errorf("error writing file %q: %v", gomod, err)
 	}
+
 	return gomod
 }
 
+// local returns a localized path for the package if it is local, as well as a
+// boolean indicating if it was found
+func local(o *Output, pkgpath string) (string, bool) {
+	cwd, err := os.Getwd()
+	if err != nil {
+		o.Errorf("error finding cwd: %v", err)
+		return "", false
+	}
+	// Hacky way of determining if we're running gomacro from a directory that
+	// likely shares a go.mod file with the pkgpath. It assumes the both the
+	// cwd and module location will include the pattern 'owner/project', which
+	// is often true.
+	// TODO: find a better way to determine if it's local
+	modParts := strings.Split(pkgpath, "/")
+	if len(modParts) > 2 &&
+		strings.Contains(strings.ToLower(cwd), strings.ToLower(strings.Join(modParts[1:3], "/"))) {
```

Comment:
I don't think this is good enough even for internal use: for example I think if you do `import "github.com/Khan/webapp/mypkg"` from `~/khan/webapp`, not to mention of course if your checkout is in a nonstandard location (I have a second worktree at `~/khan/webapp2`, although this would not be the first thing to not work great with that).

Two options come to mind:
1. implement the `import "./localpath"` syntax the maintainer suggested
2. find the go.mod for the cwd, and see what its package path is (and return true if it's a prefix of pkgpath)

Reply:
I take your point, and agree as it actually does break for my secondary branch. It may be worth pointing out that this actually does work for both the cases you specified, because both paths include the owner/project string `khan/webapp` (checked case insensitively). But it doesn't actually work for my secondary repo, `~/khan/secondary/webapp`. Given the code @StevenACoffman shared works perfectly well in that case, I think using that along with your second suggestion is the fastest way to get up and running.

As stated in the description, implementing `import "./foo"` and `import "../foo"` style paths breaks other parts of the tooling. It should be fixable, but not as quick of a way to get up and running here as just fixing the local go.mod finding.